osf/models/files.py

```diff
@@ -1,273 +1,144 @@
+            return qs.filter(provider=self.model._provider)
+        return qs
 
-    def belongs_to_node(self, node_id):
-        """Check whether the file is attached to the specified node."""
-        return self.node._id == node_id
 
-    def get_extra_log_params(self, comment):
-        return {'file': {'name': self.name, 'url': comment.get_comment_page_url()}}
+class UnableToResolveFileClass(Exception):
+    pass
 
-    def restore(self, recursive=True, parent=None):
-        """Recreate a StoredFileNode from the data in this object
-        Will re-point all guids and finally remove itself
-        :raises KeyExistsException:
-        """
-        local_field_names = self._meta.get_all_field_names()
-        intersecting_field_names = set(
-            local_field_names
-        ).intersection(set(
-            FileNode._meta.get_all_field_names()
-        ))
-
-        data = {key: getattr(self, key) for key in intersecting_field_names if
-                getattr(self, key, None) is not None}
-        if parent:
-            data['parent_id'] = parent.pk
-
-        restored = FileNode.resolve_class(self.provider, int(self.is_file))(**data)
-        if not restored.parent:
-            raise ValueError('No parent to restore to')
-        restored.save()
 
-        if recursive:
-            for child in self.children:
-                child.restore(recursive=recursive, parent=restored)
-        TrashedFileNode.remove_one(self)
-        return restored
+class DeprecatedException(Exception):
+    pass
 
 
-class StoredFileNode(CommentableMixin, OptionalGuidMixin, Taggable, ObjectIDMixin, BaseModel):
+class BaseFileNode(TypedModel, CommentableMixin, OptionalGuidMixin, Taggable, ObjectIDMixin, BaseModel):
     """
         The storage backend for FileNode objects.
         This class should generally not be used or created manually as FileNode
         contains all the helpers required.
         A FileNode wraps a StoredFileNode to provider usable abstraction layer
     """
-
     # TODO DELETE ME POST MIGRATION
     modm_model_path = 'website.files.models.base.StoredFileNode'
     modm_query = None
     migration_page_size = 10000
-    # /TODO DELETE ME POST MIGRATION
+    # /TODO DELETE ME POST MIGRATION]
+    version_identifier = 'revision'  # For backwards compatibility
+    FOLDER, FILE, ANY = 0, 1, 2
 
+    # The User that has this file "checked out"
+    # Should only be used for OsfStorage
+    checkout = models.ForeignKey('osf.OSFUser', blank=True, null=True)
     # The last time the touch method was called on this FileNode
     last_touched = NonNaiveDateTimeField(null=True, blank=True)
     # A list of dictionaries sorted by the 'modified' key
     # The raw output of the metadata request deduped by etag
     # Add regardless it can be pinned to a version or not
-    history = DateTimeAwareJSONField(default=[], blank=True)
+    _history = DateTimeAwareJSONField(default=[], blank=True)
```

Comment:
default should just be `list`. list instances are mutable.

Why is this underscored? is there special treatment of it anywhere?